api/v1alpha3/condition_consts.go

```diff
@@ -25,3 +25,34 @@ const (
 )
 
 // ANCHOR_END: CommonConditions
+
+// Conditions and condition Reasons for the Machine object
+
+// MachineConditionsCount defines the total number of conditions on the Machine object.
+const MachineConditionsCount = 2
```

Comment:
Is this expected to be the total number of conditions defined anywhere that get put onto Machines, or just the number that are being used for moving towards "ready" in the summary? We are about to run into this problem with MHC, where we're adding two more conditions to Machine. I think we shouldn't, but then we should probably clarify what this constant is. Maybe like `SummaryConditionsCount`? Not great, but something like that?

Reply:
This expected to be the total number of conditions defined anywhere that get put onto Machines; 
FYI. I'm considering changing the 1 of x behavior because it works well during provisioning but it is not ideal when the machine is up and running. But for now, this is enough for getting us started

Reply:
Name fixed

Reply:
So we will want to update this to 4 when the MHC changes get merged? Won't it always be stuck at 2/4 in the progress summary?

Reply:
Might be good to track this as a separate issue for cleanup

Reply:
+1

Reply:
I have opened https://github.com/kubernetes-sigs/cluster-api/issues/3151 to track the problem and https://github.com/kubernetes-sigs/cluster-api/pull/3152 to implement a first fix to avoid misleading messages